File: examples/source/CSharp/ex2.cs
using System ;
using CodeBase ;

namespace c4ap
{
   class ex2
   {
      [STAThread]
      static void Main(string[] args)
      {
         Code4 cb = new Code4();

         // Do not automatically open production index file
         cb.autoOpen = 0 ;
         Data4 info = new Data4( cb, "INFO.DBF" ) ;
         Index4 infoIndex = info.index("INFO") ;

         if( infoIndex.isValid() == 0 )
            Console.WriteLine( "Production index file is not opened" ) ;

         // DATA.DBF has a production index.  Open it
         cb.autoOpen = 1 ;
         Data4 data = new Data4( cb, "DATA.DBF" ) ;

         //  Some other code

         cb.closeAll( ) ;
         cb.initUndo( ) ;
      }
   }
}

File: examples/source/VB.NET/EX2.vb
Option Strict Off
Option Explicit On
Module EXAMPLE 
    'EX2.VB
	
	Public lf As String 'Line Feed
	Public fPath As String 'Full path name to data files
	
	Public Const MB_OK As Short = 0
	Public Const MB_YESNO As Short = 4
	Public Const MB_ICONQUESTION As Short = 32
	Public Const IDYES As Short = 6
	
   Sub ExCode(ByRef frm As Form1)
      'ex2 example code

      Dim db, cb, info, ind As Integer
      Dim rc As Short

      cb = code4init()

      ' Do not automatically open production index file
      rc = code4autoOpen(cb, 0)

      'Specify full path if stand-alone
      If u4switch() And &H80S Then fPath = VB6.GetPath & "\"

      info = d4open(cb, fPath & "INFO")
      ind = i4open(info, fPath & "INFO")

      If ind = 0 Then
         MsgBox("Production index file is not opened", MsgBoxStyle.OKOnly)
      End If

      ' DATA1.DBF has a production index.  Open it
      rc = code4autoOpen(cb, 1)
      db = d4open(cb, fPath & "DATA1")

      ' Some other code

      rc = code4close(cb)
      rc = code4initUndo(cb)
   End Sub
End Module